### PR TITLE
Job runner image

### PR DESCRIPTION
**Pull request by zsoltbalogh**

### CODE REVIEW

File: templates/job-runner/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ARG LABEL_NAME
+ARG LABEL_VCS_REF
+ARG LABEL_VCS_URL
+ARG LABEL_VERSION
+
+RUN apt-get install cron && \
+	apt-get clean
+
+COPY scripts/* /usr/local/bin/
+
```
Comment (gamerson) [inline templates/job-runner/Dockerfile:13]: @zsoltbalogh  regarding my slack post about redeploy "slowness", here is a quick fix
```
ADD https://github.com/krallin/tini/releases/download/v0.19.0/tini /tini
RUN chmod +x /tini
```

File: templates/job-runner/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ENTRYPOINT ["/usr/local/bin/liferay_job_runner_entrypoint.sh"]
```
Comment (gamerson) [inline templates/job-runner/Dockerfile:14]:
```suggestion
ENV TINI_VERSION v0.19.0
ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
RUN chmod +x /tini

ENTRYPOINT ["/tini", "-v", "--", "/usr/local/bin/liferay_job_runner_entrypoint.sh"]
```

Comment (gamerson) [reply templates/job-runner/Dockerfile:14]: hey @zsoltbalogh  here is a way that we can get the job-runner container to listen for SIGTERM events.  For reference: https://hynek.me/articles/docker-signals/

**Comment by zsoltbalogh** [comment]
@gamerson, can you see my changes? I'm installing tini from the ubuntu image directly so we don't need to rely on an outside source.

**Comment by gamerson** [comment]
> @gamerson, can you check my changes? I'm installing tini from the ubuntu image directly so we don't need to rely on an outside source.

Hey @zsoltbalogh  I'll check them today and let you know.  But this looks good.

File: templates/job-runner/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ARG LABEL_VCS_REF
+ARG LABEL_VCS_URL
+ARG LABEL_VERSION
+
+RUN apt-get install cron tini && \
+	apt-get clean
+
+COPY scripts/* /usr/local/bin/
+
+ENTRYPOINT ["/tini", "-v", "--", "/usr/local/bin/liferay_job_runner_entrypoint.sh"]
```
Comment (gamerson) [inline templates/job-runner/Dockerfile:14]:
```suggestion
ENTRYPOINT ["tini", "-v", "--", "/usr/local/bin/liferay_job_runner_entrypoint.sh"]
```
`tini` isn't in the root anymore so we need to remove the `/`

**Comment by zsoltbalogh** [comment]
Thank you, @gamerson ! I don't know how the image worked with the /tini, thanks for catching! I was also surprised to see that working, however the image booted up without a problem.

**Comment by zsoltbalogh** [comment]
@gamerson , @rotty3000, do you think we should use tini in our DXP image as well?

**Comment by rotty3000** [comment]
tldr; yes!

@zsoltbalogh I think it would be better long term if you migrated to using an init implementation like `tini` or `dumb-init` (either one).

FYI, not all Java applications will handle `SIGTERM` properly (the signal used by kubernetes to ask a container to shutdown gracefully). I've learned that Tomcat specifically _does_ handle it properly. However the danger with that is when you have lots of scripted orchestration (like you do) around calling the tomcat start scripts it is very easy to lose the signal propagation. This is why, regardless of the fact that Tomcat would do the right thing (if the signal reached it), I'd eliminate all doubt by just asking tomcat to shutdown directly.

**Comment by zsoltbalogh** [comment]
Thank you, @rotty3000 and @gamerson ! https://issues.liferay.com/browse/DOCKER-77

**Comment by brianchandotcom** [comment]
Ok.. on it..

**Comment by brianchandotcom** [comment]
Merged. Thank you.
View total diff: https://github.com/brianchandotcom/liferay-docker/compare/2bfe6f66c9...62f19411d1
